streams_map.go

```diff
@@ -29,26 +29,27 @@ type streamsMap struct {
 	closeErr           error
 	nextStreamToAccept protocol.StreamID
 
-	newStream newStreamLambda
+	newStream            newStreamLambda
+	removeStreamCallback removeStreamCallback
```

Comment:
It might be nicer to define a streamMapDelegate interface with these two methods, and pass the session in. Your call.